fix: Send the same country code to the config api and in the payload

corycaywood opened the pull request:
# Description 
Previously, the config API was always sent the country code found from the device locale, whereas the payload was sending the SIM card's country code or the locale if no SIM card is found. This commit changes the ConfigStore to use the same method for retrieving the country code as the payload does.

## Links
APTT-374

# Checklist
- [x] I have read the [contributing guidelines](../CONTRIBUTING.md).
- [x] I wrote/updated tests for new/changed code
- [x] I ran `./gradlew check` without errors

NemoOudeis commented on Runtime/src/main/java/com/rakuten/tech/mobile/perf/runtime/internal/RuntimeContentProvider.java:
```diff
@@ -42,26 +43,27 @@ public boolean onCreate() {
+    String locationUrlPrefix = Util.getMeta(context,
+        "com.rakuten.tech.mobile.perf.LocationUrlPrefix");
+    LocationStore locationStore = new LocationStore(context, queue, subscriptionKey,
+        locationUrlPrefix);
```
pulling the `LocationStore` out of the `if (config != null)` check will cause a bug:
if the we have no config or we decide to not track (lose coin toss vs `enablePercentage`) this will still request location from the location API -> that should not happen.

we def need to change that back

- [x] only request location if tracking is enabled

NemoOudeis commented on Core/src/main/java/com/rakuten/tech/mobile/perf/core/EnvironmentInfo.java:
```diff
@@ -9,7 +9,7 @@
 import java.util.Observer;
 
-class EnvironmentInfo implements Observer {
+public class EnvironmentInfo implements Observer {
```
I dislike making `EnvironmentInfo` public, it's not intended for use by SDK consumers.
Instead of exposing the class we could:
* move the internal classes in `Runtime` to `com.rakuten.tech.mobile.perf.core` package & use `@hide` to hide it from javadoc -> no additional exposure
* extract the country code lookup into a function and make that public -> limited exposure

- [x] reduce visibility of `EnvironmentInfo`

corycaywood replied:
I extracted this to a Util class. It seems to be a much simpler solution than what I did previously.

corycaywood committed:
fix: Send the same country code to the config api and in the payload

Previously, the config API was always sent the country code found from the device locale, whereas the payload was sending the SIM card's country code or the locale if no SIM card is found. This commit changes the ConfigStore to use the same method for retrieving the country code as the payload does.

corycaywood committed:
test: Fix async test in 'TrackerImplSpec'

Previously, this test was failing sometimes on CI

NemoOudeis commented on Core/src/test/java/com/rakuten/tech/mobile/perf/core/EnvironmentInfoSpec.java:
```diff
@@ -13,67 +13,34 @@
-  @Mock Context ctx;
+  @Mock private TelephonyManager telephonyManager;
+  @Mock private ActivityManager activityManager;
+  @Mock private Context context;
```
doesn't mockito complain about private members?

corycaywood replied:
Oh, really? I didn't see any warnings, but Android Studio warns that it can be private.

I guess all the other classes aren't using private members, so I'll change this back to match.

NemoOudeis replied:
I used to get runtime errors of mockito not being able to inject private members - maybe mockito has changed though.

NemoOudeis commented on Core/src/test/java/com/rakuten/tech/mobile/perf/core/TelephonyUtilSpec.java:
```diff
@@ -0,0 +1,90 @@
+  @Test
+  public void shouldReadCountryFromTelephonyManager() {
+    when(telephonyManager.getSimCountryIso()).thenReturn("TEST-SIM-COUNTRY");
+    assertThat(TelephonyUtil.getCountryCode(context)).isEqualTo("TEST-SIM-COUNTRY");
```
- [x] add blank line between setup and action+expectation

NemoOudeis commented on Core/src/test/java/com/rakuten/tech/mobile/perf/core/TelephonyUtilSpec.java:
```diff
@@ -0,0 +1,90 @@
+  }
+
+  @Test
+  public void shouldFallbackToReadCountryFromLocaleIfCountryIsNull() {
```
i think the naming is a bit hard to understand
> should Fallback To Read Country From Locale

so far all good
> If Country Is Null

what?

looking at the code i think it should be something like
> if sim country is null

corycaywood replied:
Yeah, that's probably a better description.

NemoOudeis commented on Core/src/test/java/com/rakuten/tech/mobile/perf/core/TelephonyUtilSpec.java:
```diff
@@ -0,0 +1,90 @@
+  @Test
+  public void shouldReadNetworkFromTelephonyManager() {
+    when(telephonyManager.getNetworkOperatorName()).thenReturn("test-network-operator");
+    assertThat(TelephonyUtil.getNetwork(context)).isEqualTo("test-network-operator");
```
- [x] add blank line between setup and action+expectation

NemoOudeis commented on Core/src/test/java/com/rakuten/tech/mobile/perf/core/TelephonyUtilSpec.java:
```diff
@@ -0,0 +1,90 @@
+  @Test
+  public void shouldFallbackToDefaultWhenTelphonyMangerIsNull() {
+    when(context.getSystemService(Context.TELEPHONY_SERVICE)).thenReturn(null);
+    assertThat(TelephonyUtil.getNetwork(context)).isEqualTo("wifi");
```
- [x] add blank line between setup and action+expectation

NemoOudeis commented on Core/src/test/java/com/rakuten/tech/mobile/perf/core/TelephonyUtilSpec.java:
```diff
@@ -0,0 +1,90 @@
+  @Test
+  public void shouldFallbackToDefaultWhenNetworkNameIsEmpty() {
+    when(telephonyManager.getNetworkOperatorName()).thenReturn("");
+    assertThat(TelephonyUtil.getNetwork(context)).isEqualTo("wifi");
```
- [x] add blank line between setup and action+expectation

NemoOudeis commented on Core/src/test/java/com/rakuten/tech/mobile/perf/core/TelephonyUtilSpec.java:
```diff
@@ -0,0 +1,90 @@
+  @Test
+  public void shouldFallbackToDefaultWhenNetworkNameIsNull() {
+    when(telephonyManager.getNetworkOperatorName()).thenReturn(null);
+    assertThat(TelephonyUtil.getNetwork(context)).isEqualTo("wifi");
```
- [x] add blank line between setup and action+expectation

NemoOudeis commented on Core/src/test/java/com/rakuten/tech/mobile/perf/core/TrackerImplSpec.java:
```diff
@@ -130,7 +130,7 @@ public Measurement getByTrackingId(int trackingId) {
       tracker.startMetric("some metric");
       Thread.sleep(100);
       tracker.prolongMetric();
-      assertThat(metric.get().endTime).isGreaterThan(beforeStart + 100);
```
lol yes 😂

NemoOudeis reviewed (changes requested):
a few minor changes in tests please
other than that LGTM
